src/main/kotlin/no/skatteetaten/aurora/boober/service/resourceprovisioning/DatabaseSchemaProvisioner.kt

```diff
@@ -14,19 +16,37 @@ import org.springframework.web.client.HttpClientErrorException
 import org.springframework.web.client.RestTemplate
 
 sealed class SchemaProvisionRequest {
-    abstract val schemaName: String
+
+    abstract val details: SchemaRequestDetails
 }
 
-data class SchemaIdRequest(val id: String, override val schemaName: String) : SchemaProvisionRequest()
+data class SchemaRequestDetails(
+    val schemaName: String,
+    val parameters: Map<String, String>,
+    // Hvis roles er tom fyll inn SCHAMA med ALL
+    val roles: Map<String, DatabasePermission>,
+    val exposeTo: Map<String, String>,
+    val flavor: DatabaseFlavor,
+    val affiliation: String
+)
```

Comment:
Jeg lurer på om jeg nok syntes at detaljer fra AuroraConfig/boober-domenet lekker for mye over i database-apiet. Fra dbh sin side så trenger den bare vite hvilke brukere den ønsker at vi skal opprette og hvilke rettigheter disse skal ha.

Foreslår noe i denne duren;

{
  "schemaName": "løakjfdaøsldfjalsfj",
  "parameters": {
    "instanceName": "sirius"
  },
  "engine": "postgres",
  "users": {
    "readonly": "r",
    "app": "rw",
    "migration": "a",
    "serg": "rw"
  }
}

evt.

{
  "schemaName": "løakjfdaøsldfjalsfj",
  "parameters": {
    "instanceName": "sirius"
  },
  "engine": "postgres",
  "users": [
    {"name": "readonly", "role": "r"},
    {"name": "app", "role": "rw"},
    {"name": "migration", "role": "a"},
    {"name": "serg", "role": "rw"}
  ]
}

Jeg heller mot den siste siden den er mer future proof (mulig å utvide med nye felt, f.eks. dersom man ønsker sette passord eller restrictions (bare mellom 8 og 16) ell.). Det blir også lettere å lese koden som parser siden man slipper forholde seg til en Map<String, String> og gjette på hva key og value faktisk betyr. Løsningen med users som Map<String, String> har bare fordelen at den er litt mer kompakt - utover det stort sett bare ulemper. Kompakthet er ikke et altoverskyggende mål i maskin-til-maskin-apier, syntes jeg.

Reply:
Enig. Dette kan jeg fikse. Vi går for den siste som du anbefaler.

Reply:
Hvor skal tilgangskontroll til skjema ligge? Per idag er det pip åpent. Så lenge man vet en ID som man kan finne i en fil så kan man hente noe, jeg synes ikke dette er noe vi egentlig vil ha. 

Forslaget mitt går på at dette ansvaret skal ligge i DBH. Når man lager skjema så sier man hvilke roller man skal ha, hvilken tilgang de skal ha og om andre foretningsgrupper/tilhørigheter skal kunne få hente dem. Når man spør etter skjema må man da alltid sende med sin tilhørighet, til og med for id kall. Man kan også sende med en eller flere roller, sender man ikke med roller får man alle rollene man har lov til å se.